lib/Doctrine/ORM/Cache/DefaultCacheFactory.php

```diff
@@ -178,6 +181,19 @@ public function buildQueryCache(EntityManagerInterface $em, $regionName = null)
      */
     public function buildCollectionHydrator(EntityManagerInterface $em, array $mapping)
     {
+        if ($mapping['cache']) {
```

Comment:
This condition is always true. Method `buildCollectionHydrator()` is only called through `AbstractCollectionPersisters` in Cache namespace, which gets decided inside of UnitOfWork by checking this condition already.